resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterManager.scala

```diff
@@ -138,6 +133,31 @@ private[spark] class KubernetesClusterManager extends ExternalClusterManager wit
       podsPollingEventSource)
   }
 
+  private[k8s] def makeExecutorPodsAllocator(sc: SparkContext, kubernetesClient: KubernetesClient,
+      snapshotsStore: ExecutorPodsSnapshotsStore) = {
+    val executorPodsAllocatorName = sc.conf.get(KUBERNETES_ALLOCATION_PODSALLOCATOR) match {
+      case "statefulset" =>
+        "org.apache.spark.scheduler.cluster.k8s.StatefulsetPodsAllocator"
+      case "direct" =>
+        "org.apache.spark.scheduler.cluster.k8s.ExecutorPodsAllocator"
```

Comment:
Nit: Possibly consider using some named constants or `classOf` with `.toString`? Can be left as a follow up too.

Reply:
Oh hells yeah, good call.